src/main/java/bgu/spl/mics/application/passiveObjects/Agent.java

```diff
@@ -1,71 +1,90 @@
 package bgu.spl.mics.application.passiveObjects;
 
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
 /**
  * Passive data-object representing a information about an agent in MI6.
- * You must not alter any of the given public methods of this class. 
+ * You must not alter any of the given public methods of this class.
  * <p>
  * You may add ONLY private fields and methods to this class.
  */
 public class Agent {
 
-	public Agent(String s, String a) {
-	}
+    private AtomicReference<String> _name;
+    private AtomicReference<String> _serial;
+    private AtomicBoolean _available;
+
+    public Agent(String name, String serial) {
```

Comment:
I think serial should come before name because it is the main field of the agent

Reply:
Done